java.base/src/main/java/java/lang/ProcessHandleImpl$Info$_patch.java

```diff
@@ -0,0 +1,237 @@
+ *
+ * ------
+ * Copyright (c) 2001, 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ * ------
+ *
+ * This file may contain additional modifications which are Copyright (c) Red Hat and other
+ * contributors.
+ */
+package java.lang;
+
+import static org.qbicc.runtime.CNative.*;
+import static org.qbicc.runtime.bsd.SysProc.*;
+import static org.qbicc.runtime.bsd.SysSysctl.*;
+import static org.qbicc.runtime.linux.Unistd.*;
+import static org.qbicc.runtime.posix.Errno.*;
+import static org.qbicc.runtime.posix.Pwd.*;
+import static org.qbicc.runtime.posix.String.*;
+import static org.qbicc.runtime.posix.SysResource.*;
+import static org.qbicc.runtime.posix.SysTime.*;
+import static org.qbicc.runtime.posix.SysTypes.*;
+import static org.qbicc.runtime.posix.Unistd.*;
+import static org.qbicc.runtime.stdc.Errno.*;
+import static org.qbicc.runtime.stdc.Stddef.*;
+import static org.qbicc.runtime.stdc.Stdint.*;
+import static org.qbicc.runtime.stdc.Stdlib.*;
+import static org.qbicc.runtime.stdc.String.*;
+
+import org.qbicc.rt.annotation.Tracking;
+import org.qbicc.runtime.Build;
+import org.qbicc.runtime.patcher.Patch;
+import org.qbicc.runtime.patcher.Add;
+import org.qbicc.runtime.patcher.Replace;
+
+@Patch("java.lang.ProcessHandleImpl$Info")
+@Tracking("src/java.base/unix/native/libjava/ProcessHandleImpl_unix.c")
+@Tracking("src/java.base/windows/native/libjava/ProcessHandleImpl_win.c")
+class ProcessHandleImpl$Info$_patch {
+    // alias
+    String command;
+    String commandLine;
+    String[] arguments;
+    long startTime;
+    long totalTime;
+    String user;
+
+    @Replace
+    static void initIDs() {}
+
+    @Add
+    static pid_t getParentPidAndTimings(pid_t pid, int64_t_ptr totalTime, int64_t_ptr startTime) {
```

Comment:
This method maybe belongs in some helper class since it is invoked from native methods in both ProcessHandleImpl and ProcessHandleImpl$Info.  It has an `os_` prefix in the JDK version